beard/metrics/clustering.py

```diff
@@ -19,9 +20,28 @@
 from operator import mul
 from itertools import groupby
 
+from sklearn.metrics import silhouette_score as sklearn_silhouette_score
 from sklearn.metrics.cluster.supervised import check_clusterings
 
 
+def silhouette_score(X, labels, metric="precomputed"):
+    """Compute silhouette_score with an extension of number of clusters condition.
+
+    The condition is that Silhouette Coefficent is only defined if number of
+    clusters is 1 < n_clusters < n_samples. What the extension does here is
+    simply accepts n_clusters out of that boundary and returns the minimum
+    possible score for such cases.
+
+    Please check sklearn.metrics.silhouette_score for the documentation
+    """
+    n_samples = X.shape[0]
+    n_clusters = len(np.unique(labels))
+    if 1 < n_clusters < n_samples:
+        return sklearn_silhouette_score(X, labels, metric)
+    else:
+        return -1
+
+
```

Comment:
Since you are adding the silhouette score in this PR, you could also add the others directly now I think.